assets/src/editor/blocks/video-promo-container/view.js

```diff
@@ -0,0 +1,125 @@
+/**
```

Comment:
For readability, I'd maybe consider restructuring this a little to move the functions out of the document.addEventListener and the forEach loop. Maybe something like

```js
const setVideoHeight = ( video ) => {
	//... 
}

// And other functions

document.addEventListener( 'DOMContentLoaded', () => {
	const containers = document.querySelectorAll( '.wp-block-shiro-video-promo-container' );

	containers.forEach( container => {
		const video = container.querySelector( 'video' );

		if ( ! video ) {
			return;
		}

		setVideoHeight( video );
		window.addEventListener( 'resize', () => setVideoHeight( video) );

		video.addEventListener( 'play', () => scrollVideoIntoView( video, container ) );

		// etc

```

Works fine as is, but just a thought.